pasqal_cloud/workload.py

```diff
@@ -52,3 +54,19 @@ def cancel(self) -> Dict[str, Any]:
         workload_rsp = self._client._cancel_workload(self.id)
         self.status = workload_rsp.get("status", "CANCELED")
         return workload_rsp
+
+    @validator("result")
+    def result_link_to_result(
+        cls, result: Optional[Dict[str, Any]], values: Dict[str, Any]
+    ) -> Optional[Dict[str, Any]]:
+        result_link: Optional[str] = values["result_link"]
+        if result or not result_link:
+            return result
+        try:
+            res = requests.get(result_link)
+            data = res.json()
+        except Exception:
+            raise ValueError("Invalid result link.")
```

Comment:
The link could be valid but there could be a networking issue. I suggest raising a `Error while downloading workload results` error instead.

Moreover, can we create a custom exception for this like in #106 ?